src/neo/Wallets/Wallet.cs

```diff
@@ -405,38 +428,56 @@ public bool Sign(ContractParametersContext context)
             foreach (UInt160 scriptHash in context.ScriptHashes)
             {
                 WalletAccount account = GetAccount(scriptHash);
-                if (account is null) continue;
 
-                // Try to sign self-contained multiSig
+                if (account != null)
+                {
+                    // Try to sign self-contained multiSig
 
-                Contract multiSigContract = account.Contract;
+                    Contract multiSigContract = account.Contract;
 
-                if (multiSigContract != null &&
-                    multiSigContract.Script.IsMultiSigContract(out int m, out ECPoint[] points))
-                {
-                    foreach (var point in points)
+                    if (multiSigContract != null &&
+                        multiSigContract.Script.IsMultiSigContract(out int m, out ECPoint[] points))
+                    {
+                        foreach (var point in points)
+                        {
+                            account = GetAccount(point);
+                            if (account?.HasKey != true) continue;
+                            KeyPair key = account.GetKey();
+                            byte[] signature = context.Verifiable.Sign(key);
+                            fSuccess |= context.AddSignature(multiSigContract, key.PublicKey, signature);
+                            if (fSuccess) m--;
+                            if (context.Completed || m <= 0) break;
+                        }
+                    }
+                    else if (account.HasKey)
```

Comment:
@erikzhang If the account it's watch-only we can't sign

Reply:
Why add watch-only account to the wallet?

Reply:
Because you can track the balance of the smart contract

Reply:
Fixed.